lib/dealership.rb

```diff
@@ -0,0 +1,86 @@
+class Dealership
+
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def has_inventory?
+    @inventory.length >= 1
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def cars_by_make(make)
+    cars_by_make_catcher = []
+    @inventory.each do |car|
+      if car.make == make
+        cars_by_make_catcher << car
+      end
+    end
+    cars_by_make_catcher
+  end
+
+  def total_value
+    total_inventory_value = 0
+    @inventory.each do |car|
+      total_inventory_value += car.total_cost
+    end
+    total_inventory_value
+  end
+
+  def details
+    dealer_details = {
+      "total_value" => total_value,
+      "address" => @address
+    }
+  end
+
+  def average_price_of_car
+    average_cost = total_value / inventory_count
+    format_number(average_cost)
+  end
+
+  def format_number(number)
+    group_nums = number.to_s.chars.to_a.reverse.each_slice(3) #
+    group_nums.map(&:join).join(',').reverse
+  end
```

Comment:
This works! It can be nice to comment where you found this to give credit to the original source and point others to the article if they want to read more about it. Sometimes that is nice when you come across something a little tricky in source code and to credit the source.